cpp/rrtmgp_conversion.h

```diff
@@ -398,6 +398,89 @@ sum(const KView& view)
   return rv;
 }
 
+// MemPool singleton
+struct MemPoolSingleton
+{
+  using DeviceMemPool = typename Kokkos::MemoryPool<typename DefaultDevice::execution_space>;
+  using HostMemPool   = typename Kokkos::MemoryPool<typename HostDevice::execution_space>;
+
+ public:
+  inline static DeviceMemPool s_device_mem_pool;
+  inline static HostMemPool   s_host_mem_pool;
+
+  static void init(
+    const size_t dev_capacity, const size_t dev_min_block_size, const size_t dev_max_block_size, const size_t dev_super_block_size,
+    const size_t hst_capacity, const size_t hst_min_block_size, const size_t hst_max_block_size, const size_t hst_super_block_size)
+  {
+    static bool is_init = false;
+    RRT_REQUIRE(!is_init, "Multiple MemPoolSingleton inits");
+    s_device_mem_pool = DeviceMemPool(typename DefaultDevice::memory_space(), dev_capacity, dev_min_block_size, dev_max_block_size, dev_super_block_size);
+    //s_host_mem_pool = HostMemPool(typename HostDevice::memory_space(), hst_capacity, hst_min_block_size, hst_max_block_size, hst_super_block_size);
+  }
+
+  template <typename T>
+  static inline
+  T* alloc(const int num) noexcept
+  {
+    T* rv;
+#ifdef KOKKOS_ENABLE_CUDA
+    uint64_t addr = 0;
+    DeviceMemPool pool_temp = s_device_mem_pool;
+    Kokkos::parallel_reduce(1, KOKKOS_LAMBDA(size_t, uint64_t& ptr) {
+      ptr = reinterpret_cast<uint64_t>(pool_temp.allocate(num * sizeof(T)));
```

Comment:
It was a bit annoying that you had to be in a kernel in order to alloc on GPU

Reply:
That means we are not using the tool for what it was intended. Did you check with the kk devs to make sure we won't get errors down the road if we use it the way we do?

Reply:
Also, what about HIP? I'm assuming the same issue will happen there?

Reply:
The Kokkos pool is essentially for scratch space, I believe, although I might be wrong. But that would explain needing to be in a kernel. GATOR I think just uses global memory, which IMO is fine. So you could make a GATOR-like memory-pool utility. See my other comment about alloc'ing the big buffer in that approach.

Reply:
From the Kokkos slack channel:
```
Damien: MemoryPool was not intended to be user facing
Damien: It is used in tasking which is not actively maintained
Me: Thanks, Damien. Is there some other Kokkos utility I should be using if I want a pool? It helped my performance a ton
Damien: I recall hearing about people using SICM from LANL or Umpire at LLNL
Damien: I guess my point is if you depend on that functionality you should lobby for something more sustainable. Maybe other users share your needs
Damien: We don’t test that stuff on AMD or Intel GPUs
```